### PR TITLE
added torch version of tabular neural network to complement mxnet

### PR DESCRIPTION
**Pull request by jwmueller**
There are now two tabular neural network models, which can be specified by `hyperparameters` keys:
`NN_MXNET` and `NN_TORCH`. The old key `NN` is now invalid with a deprecation warning. Both DL frameworks are lazy imported such that neither is necessary to run autogluon.tabular.

In the default hyperparameter config, both `NN_MXNET` and `NN_TORCH` are present for now in the set of models trained by default so they are easier to compare.  In all other hyperparameter-configs, I just kept `NN_MXNET` and did not add the new `NN_TORCH` (can just add this key down the road easily). `NN_TORCH` is also added to the default set of student models used in distillation to accompany `NN_MXNET`.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1489-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/3/index.html

**Comment by szha** [comment]
Job PR-1489-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/5/index.html

**Comment by szha** [comment]
Job PR-1489-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/6/index.html

**Comment by szha** [comment]
Job PR-1489-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/7/index.html

**Comment by szha** [comment]
Job PR-1489-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/8/index.html

**Comment by szha** [comment]
Job PR-1489-9 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/9/index.html

**Comment by szha** [comment]
Job PR-1489-10 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/10/index.html

**Comment by szha** [comment]
Job PR-1489-11 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/11/index.html

**Comment by szha** [comment]
Job PR-1489-12 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1489/12/index.html

**Review by Innixma — APPROVED** [review]
LGTM! Benchmark results show similar accuracy but ~4-5x faster inference speed (~10x+ in online inference).

There is still work to be done regarding documentation + tuning of the default parameters, but this is generally in a solid enough state to merge.
